### PR TITLE
Added a Query builder through the cakephp/database package

### PR DESCRIPTION
**Pull request by lorenzo**
Support for a more feature reach query builder has been asked for
multiple times. Creating query builders is difficult, so better use
one that already exists.

### CODE REVIEW

**Comment by lorenzo** [comment]
Merging this as there no feedback for over a week and would like to make a release soon

**Comment by rquadling** [comment]
Any reason why the notes about `change()` have been removed from the documentation? Doesn't seem to relate to this feature.

File: docs/migrations.rst [hunk 1]
```diff
@@ -55,8 +55,6 @@ Phinx automatically creates a skeleton migration file with a single method:
              *    addIndex
              *    addForeignKey
              *
-             * Remember to call "create()" or "update()" and NOT "save()" when working
```
Comment (rquadling) [inline docs/migrations.rst]: Why has this been removed?

File: docs/migrations.rst [hunk 2]
```diff
@@ -125,13 +123,6 @@ Please be aware that when a ``change`` method exists, Phinx will automatically
 ignore the ``up`` and ``down`` methods. If you need to use these methods it is
 recommended to create a separate migration file.
 
-.. note::
```
Comment (rquadling) [inline docs/migrations.rst]: Why has this been removed?

**Comment by rquadling** [comment]
Ah. I'm late on the merge, but still, those notes are important.

File: docs/migrations.rst (hunk 1 above)
Comment (lorenzo) [reply docs/migrations.rst]: I forgot to remove them in my previous PR, this is not required anymore

File: docs/migrations.rst (hunk 2 above)
Comment (lorenzo) [reply docs/migrations.rst]: Because it is not true anymore after my previous PR

Comment (rquadling) [reply docs/migrations.rst]: Ha! I'll shut up then!

**Pull request merged**